### PR TITLE
fix(types): add delegated gateway type

### PR DESCRIPTION
**Pull request by dtfiedler**

### CODE REVIEW

**Review by fedellen — APPROVED** [review]

File: src/contract-state.ts [hunk 1]
```diff
@@ -60,6 +61,12 @@ export type Gateway = {
   weights: ObserverWeights;
 };
 
+export type GatewayDelegate = {
```
Comment (kunstmusik) [inline src/contract-state.ts:64]: Could you also modify Gateway to have:

```
delegtes: Record<string, GatewayDelegate>
```

Comment (dtfiedler) [reply src/contract-state.ts:64]: ah, good catch

Comment (dtfiedler) [reply src/contract-state.ts:64]: [ac7e924](https://github.com/ar-io/ar-io-sdk/pull/86/commits/ac7e924f24ba6133b2917de8416c37bf6560cdd4)

**Pull request merged**